HW05.Rmd

```diff
@@ -0,0 +1,159 @@
+
+```{r question 1 data}
+vorinostat <- cancer_auc %>% 
+# filter AUC values and cancer type of vorinostat treated experiments
+  select(area_under_curve, cpd_name, cancer_type) %>% 
+  filter(cpd_name == "vorinostat") %>% 
+  drop_na(cancer_type) %>% 
+# calculate average AUC values
+  group_by(cancer_type) %>% 
+  summarize(AUC = round(mean(area_under_curve), digits = 2)) %>% 
+  arrange(AUC) %>% 
+  mutate(cancer_type = factor(cancer_type, levels = cancer_type))
+```
+
+```{r question 1 graph}
+# plot
+ggplot(vorinostat, aes(x = cancer_type, y = AUC)) +
+  geom_col(fill = "#00CDFF") +
+# add labels
+  labs(x = "Cancer Type", 
+       y = "Area Under Curve", 
+       title = "AUC of Different Cancer Types Treated by Vorinostat") +
+  geom_text(aes(label = AUC), vjust = -0.5, size = 2.5) +
+# adjust theme
+  theme_classic() +
+  theme(plot.title = element_text(hjust = 0.5),
+        axis.text.x = element_text(angle = -30, hjust = 0)) 
+```
+
+```{r}
+print(paste("The cancer type having the lowest AUC values to the compound vorinostat is ", gsub(as.character(vorinostat$cancer_type)[1], pattern = "_", replacement = " "), ".", sep = ""))
+```
+
+### Question 2
+
+Which compound is the prostate cancer cell line 22RV1 most sensitive to?
+
+```{r question 2 data}
+prostate_cancer <- cancer_auc %>% 
+  select(area_under_curve, cpd_name, ccl_name) %>% 
+  filter(ccl_name == "22RV1") %>% 
+# calculate average AUC values of each compound
+  group_by(cpd_name) %>% 
+  summarize(AUC = round(mean(area_under_curve), digits = 2)) %>% 
+  arrange(AUC) 
+```
+
+```{r question 2 answer}
+print(paste("The prostate cancer cell line 22RV1 is most sensitive to ", prostate_cancer[1, 1], ".", sep = ""))
+```
+
+
+### Question 3
+
+For the 10 compounds that target EGFR, which of them has (on average) the lowest AUC values in the breast cancer cell lines?
+
+```{r question 3 data}
+EGFR <- cancer_auc %>% 
+# filter AUC values of breast cancer cell lines treated by EGFR targeting compounds
+  select(area_under_curve, cpd_name, gene_symbol_of_protein_target, ccl_name, cancer_type) %>% 
+  filter(cancer_type == "breast", grepl("EGFR", gene_symbol_of_protein_target)) %>% 
+# calculate average AUC values of each cell line
+  group_by(cpd_name, ccl_name) %>% 
+  summarize(AUC_ccl = mean(area_under_curve)) %>% 
+# calculate average AUC values of each compound
+  summarize(AUC = round(mean(AUC_ccl), digits = 2)) %>% 
+  arrange(AUC) %>% 
+  mutate(cpd_name = factor(cpd_name, levels = cpd_name))
+```
+
+```{r question 3 graph}
+# plot
+ggplot(EGFR, aes(x = cpd_name, y = AUC)) +
+  geom_col(fill = "#00CDFF") +
+# add labels
+  labs(x = "Compound", 
+       y = "Area Under Curve", 
+       title = "AUC of Breast Cancer Cell Lines Treated by EGFR Targeting Compounds") +
+  geom_text(aes(label = AUC), vjust = -0.5, size = 2.5) +
+# adjust theme
+  theme_classic() +
+  theme(plot.title = element_text(hjust = 0.5),
+        axis.text.x = element_text(angle = -30, hjust = 0))
```

Comment:
My thoughts are the same as in Question 1. Good, straightforward use of tidyverse to prepare the data for visualization and this plot really does the job.